test/core/utils.js

```diff
@@ -274,464 +274,518 @@ describe("utils", function() {
-        value: "not a file"
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( ["Value must be a file"] )
-
-      // valid (empty) file
-      param = fromJS({
-        required: false,
-        type: "file",
-        value: undefined
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-
-      // valid file
-      param = fromJS({
-        required: false,
-        type: "file",
-        value: new win.File()
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-    })
-
-    it("validates required arrays", function() {
-      // invalid (empty) array
-      param = fromJS({
-        required: true,
-        type: "array",
-        value: []
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( ["Required field is not provided"] )
-
-      // invalid (not an array)
-      param = fromJS({
-        required: true,
-        type: "array",
-        value: undefined
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( ["Required field is not provided"] )
-
-      // invalid array, items do not match correct type
-      param = fromJS({
-        required: true,
-        type: "array",
-        value: [1],
-        items: {
-          type: "string"
-        }
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [{index: 0, error: "Value must be a string"}] )
-
-      // valid array, with no 'type' for items
-      param = fromJS({
-        required: true,
-        type: "array",
-        value: ["1"]
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-
-      // valid array, items match type
-      param = fromJS({
-        required: true,
-        type: "array",
-        value: ["1"],
-        items: {
-          type: "string"
-        }
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-    })
-
-    it("validates optional arrays", function() {
-      // valid, empty array
-      param = fromJS({
-        required: false,
-        type: "array",
-        value: []
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-
-      // invalid, items do not match correct type
-      param = fromJS({
-        required: false,
-        type: "array",
-        value: ["number"],
-        items: {
-          type: "number"
-        }
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [{index: 0, error: "Value must be a number"}] )
-
-      // valid
-      param = fromJS({
-        required: false,
-        type: "array",
-        value: ["test"],
-        items: {
-          type: "string"
-        }
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-    })
 
-    it("validates required booleans", function() {
```

Comment:
Add new OAS3 spec tests here. Rest of the diff is because of indentation.

Reply:
https://github.com/swagger-api/swagger-ui/pull/3751/files?w=1 gives whitespace-free diffs 😉

Reply:
should be the default!